backend/tests/mocks/isar_requests_mock.py

```diff
@@ -0,0 +1,57 @@
+from typing import List
+
+from requests import Response
+
+from flotilla.services.isar import IsarRequestsInteface
+
+
+def default_start_mission_response() -> Response:
+    response: Response = Response()
+    response.status_code = 200
+
+    default_json = {
+        "message": "Mission started",
+        "started": True,
+        "mission_id": "KAAWILLIAM03022022133849308",
+    }
+    response.json = lambda: default_json
+    return response
+
+
+def default_stop_response() -> Response:
+    response: Response = Response()
+    response.status_code = 200
+
+    default_json = {"message": "Mission stopping", "stopped": True}
+    response.json = lambda: default_json
+    return response
+
+
+class IsarRequestsMock(IsarRequestsInteface):
+    def __init__(self) -> None:
+        self.start_mission_responses: List[Response] = []
+        self.stop_responses: List[Response] = []
+
+        self.default_start_mission_reponse = default_start_mission_response()
+        self.default_stop_response = default_stop_response()
+
+    def start_mission(self, host: str, port: int, mission_id: int) -> Response:
+        if self.start_mission_responses:
+            return self.start_mission_responses.pop(0)
+        return self.default_start_mission_reponse
+
+    def stop(self, host: str, port: int) -> Response:
+        if self.stop_responses:
+            return self.stop_responses.pop(0)
+        return self.default_stop_response
+
+    def add_start_mission_responses(self, responses: List[Response]) -> None:
+        self.start_mission_responses += responses
+
+    def add_stop_responses(self, responses: List[Response]) -> None:
+        self.stop_responses += responses
```

Comment:
I do not see why we need this functionality in the mock. Is it not enough to mock `start` and `stop`?

Reply:
So you can be able to add your custom responses if you want to test how the api response to a particular response from ISAR